### PR TITLE
Clean up strange uses of getAnalysisIfAvailable

### PR DESCRIPTION
**Pull request by jayfoad**
After a pass calls addRequired<X>() it is strange to call
getAnalysisIfAvailable<X>() because analysis X should always be
available. Use getAnalysis<X>() instead.

### CODE REVIEW

**Review by nikic — APPROVED** [review]
LGTM

GVN seems to have different behavior regarding LoopInfo in the LegacyPM and NewPM, but that's an orthogonal issue. I'll check whether always requiring LoopInfo in GVN has any compile-time impact, as I don't think this is supposed to be an optional analysis there (otherwise loop load PRE is not going to work reliably).

**Review by bjope — COMMENTED** [review]
I guess I don't know how pull requests and reviewing works in github. I actually added 3 comments on this patch a several days (or weeks) ago. But turns out that they were "pending" because I had only "started review" and not found the place to "submit review".

So sorry for some late comments here. Since this has been pushed already, then I guess you may ignore them.

**Comment by jayfoad** [comment]
> I guess I don't know how pull requests and reviewing works in github. I actually added 3 comments on this patch a several days (or weeks) ago. But turns out that they were "pending" because I had only "started review" and not found the place to "submit review".

For that reason I usually click "add single comment" instead of "start a review".

**Comment by nikic** [comment]
> GVN seems to have different behavior regarding LoopInfo in the LegacyPM and NewPM, but that's an orthogonal issue. I'll check whether always requiring LoopInfo in GVN has any compile-time impact, as I don't think this is supposed to be an optional analysis there (otherwise loop load PRE is not going to work reliably).

I've checked, and LoopInfo is always available in pipeline positions where GVN runs, so this effectively has no impact. I've made it a required analysis for the NewPM as well in https://github.com/llvm/llvm-project/commit/fa5884770ab240139f33a8b2be801673ca21b9c0.

File: llvm/lib/Transforms/Scalar/GVN.cpp
Comment (jayfoad) [reply llvm/lib/Transforms/Scalar/GVN.cpp:3289]: I prefer not to leave it as is - I think either we should get the analysis with `getAnalysis` or we should remove the `addRequired`, so that they are consistent with each other.

File: llvm/lib/CodeGen/VirtRegMap.cpp
Comment (jayfoad) [reply llvm/lib/CodeGen/VirtRegMap.cpp:264]: Thanks! Fixed in 05c16f40c9de93c181d45ec718de5487380f0514

File: llvm/lib/Target/AArch64/AArch64ConditionalCompares.cpp
Comment (jayfoad) [reply llvm/lib/Target/AArch64/AArch64ConditionalCompares.cpp:938]: Everything seems to work if I change `AArch64ConditionalCompares` to not require `MachineLoopInfo`. But I think I'll leave it to the AArch64 maintainers like you suggest.

**Comment by bjope** [comment]
> > I guess I don't know how pull requests and reviewing works in github. I actually added 3 comments on this patch a several days (or weeks) ago. But turns out that they were "pending" because I had only "started review" and not found the place to "submit review".
> 
> For that reason I usually click "add single comment" instead of "start a review".

Thanks. I'll try to remember that.

If they for example had named the buttons "Post comment directly" and "Save as draft", then it has been more obvious.
I also do not know how to find all my pending "draft reviews" (unless I'm already "assigned" or "requested as reviewer") as the https://github.com/pulls is just crap compared to my old dashboard in phabricator.

**Pull request merged**
